bin/exam.py

```diff
@@ -855,6 +856,7 @@ def toxml(self):
         part.find('correctanswer').text = strcons(self.answer)
 
         part.find('case').attrib = {'sensitive': strcons_fix(self.caseSensitive), 'partialcredit': strcons_fix(self.partialCredit)+'%'}
+        part.find('matchmode') = ## ????
```

Comment:
In this method, we create an XML element with the following structure:

        <part>
        - <displayanswer>
        - <correctanswer>
        - <case>

The content of the `correctanswer` tag is the string we want to match against (currently always a regular expression). To specify how the correctanswer should be used, I think it makes sense to store `matchMode` in an attribute of this element.
So your line here should be something like:

        part.find('correctanswer').attrib = { 'mode': strcons(self.matchMode) }

(by the way, the `strcons` thing is a wrapper to make sure we get a unicode string on both Python 2 and 3)